src/devices/Board/I2cBusExtensions.cs

```diff
@@ -91,6 +91,10 @@ public static (List<int> FoundDevices, int LowestAddress, int HighestAddress) Pe
                 {
                     // Ignore and continue
                 }
+                finally
+                {
+                    bus.RemoveDevice(addr);
+                }
```

Comment:
What hardware are you running on? At least `UnixI2cDevice.Dispose()` already calls `RemoveDevice`,  so this shouldn't be necessary.

Reply:
agreed, we should be removing device from the bus in the I2cDevice.Dispose - I'm guessing FT232H might be missing that (assuming that's what you're using)

Reply:
Seems that's already there
https://github.com/dotnet/iot/blob/main/src/devices/Ft232H/Ft232HI2c.cs#L50

might be worth debugging... `using` should be taking care of the RemoveDevice since it calls Dispose

Reply:
I think the problem is here:
https://github.com/dotnet/iot/blob/main/src/devices/Ft232H/Ft232HI2cBus.cs#L36-L41

we're adding to a list even when error occurs

Reply:
(might be worth to review other similar implementations and see what they do)

Reply:
discussion continued in https://github.com/dotnet/iot/pull/2032#issuecomment-1423432449